src/app/portlets/dot-edit-content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.ts

```diff
@@ -55,6 +54,39 @@ export class DotEditPageToolbarComponent implements OnInit {
                 };
             });
         }
+
+        this.stateSelected = this.page.locked ? PageMode.EDIT : PageMode.PREVIEW;
+    }
+
+    /**
+     * Copy url to clipboard
+     *
+     * @returns {boolean}
+     * @memberof DotEditPageToolbarComponent
+     */
+    copyUrlToClipboard(): boolean {
+        /*
+            Aparently this is the only crossbrowser solution so far. If we do this in another place we might have
+            to include an npm module.
+        */
+        const txtArea = document.createElement('textarea');
+
+        txtArea.style.position = 'fixed';
+        txtArea.style.top = '0';
+        txtArea.style.left = '0';
+        txtArea.style.opacity = '0';
+        txtArea.value = this.page.pageUri;
+        document.body.appendChild(txtArea);
+        txtArea.select();
+
+        try {
+            return document.execCommand('copy');
+        } catch (err) {
+            console.log('Oops, unable to copy');
```

Comment:
shouldn't we show a error message in this case?

Reply:
shouldn't we have a Logger Service?

Reply:
We do have one (need to replace it because it's not AoT ready) but we shouldn't log this, we have a notification system for it.